packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -332,6 +333,9 @@ class FlutterErrorDetails extends Diagnosticable {
   /// otherwise the summary is inferred from the string representation of the
   /// exception.
   DiagnosticsNode get summary {
+    if (kReleaseMode) {
+      return DiagnosticsNode.message(exceptionAsString());
```

Comment:
To keep the string based display consistent perhaps you want
exceptionAsString().split("\n")[0].trimLeft()
You could add this as a private helper as it is used in two places.